### PR TITLE
Add have_implicit_order_column matcher

### PR DESCRIPTION
**Pull request by yaorlov**
Add `have_implicit_order_column` matcher to test rails 6 [implicit_order_column](https://github.com/rails/rails/pull/34480)

### CODE REVIEW

File: spec/support/acceptance/helpers/step_helpers.rb [hunk 1]
```diff
@@ -125,5 +125,9 @@ def add_spring_to_project
 
       add_gem 'spring-commands-rspec'
     end
+
+    def sqlite3_version
+      rails_version >= '6' ? '~> 1.4' : '~> 1.3.6'
```
Comment (mcmire) [inline spec/support/acceptance/helpers/step_helpers.rb]: Ah, good catch on this. 👍

File: Appraisals [hunk 2]
```diff
@@ -113,24 +113,27 @@ end
 
 if Gem::Requirement.new('>= 2.5.0').satisfied_by?(Gem::Version.new(RUBY_VERSION))
   appraise 'rails_6_0' do
-    instance_eval(&shared_dependencies)
+    instance_eval(&shared_jruby_dependencies)
```
[the remark this answers is not in the document]
Comment (yaorlov) [reply Appraisals]: Done

File: lib/shoulda/matchers/active_record/have_implicit_order_column.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        def description
+          "have implicit_order_column assigned to #{@column}"
+        end
+
+        private
+
+        def column_exists?
+          matcher = HaveDbColumnMatcher.new(@column)
+          return true if matcher.matches?(@subject)
```
[the remark this answers is not in the document]
Comment (yaorlov) [reply lib/shoulda/matchers/active_record/have_implicit_order_column.rb]: Done

File: spec/unit/shoulda/matchers/active_record/have_implicit_order_column_spec.rb [hunk 4]
```diff
@@ -0,0 +1,132 @@
+require 'unit_spec_helper'
+
+describe Shoulda::Matchers::ActiveRecord::HaveImplicitOrderColumnMatcher, type: :model do
+  if active_record_supports_implicit_order_column?
+    context 'when implicit_order_column is defined for the column' do
+      let(:defined_model) { model(created_at: :timestamp) }
```
[the remark this answers is not in the document]
Comment (yaorlov) [reply spec/unit/shoulda/matchers/active_record/have_implicit_order_column_spec.rb]: Done. No more `let`s and `before`s

**Comment by cprodhomme** [comment]
any news @mcmire ? :)

i really need this PR to test `implicit_order_column` 😃

**Comment by mcmire** [comment]
Yup, so sorry @cprodhomme. I think this is ready to go but I will take another look at this later today.

**Comment by mcmire** [comment]
Thanks for this PR! I took some liberties in bringing this up to current standards and in tweaking the error message a bit. It's in master now.

**Pull request merged**